Dockerfile

```diff
@@ -0,0 +1,36 @@
+#=============== Build Container ===================
+FROM golang:1.17-stretch
+
+ARG GIT_COMMIT_SHORT="dev"
+ARG KUBECTL_VERSION=1.23.0
+
+# On CyberArk dev laptops, golang module dependencies are downloaded with a
+# corporate proxy in the middle. For these connections to succeed we need to
+# configure the proxy CA certificate in build containers.
+#
+# To allow this script to also work on non-CyberArk laptops where the CA
+# certificate is not available, we copy the (potentially empty) directory
+# and update container certificates based on that, rather than rely on the
+# CA file itself.
+ADD build_ca_certificate /usr/local/share/ca-certificates/
+RUN update-ca-certificates
+
+RUN mkdir -p /work
+WORKDIR /work
+
+ENV GOOS=linux \
+    GOARCH=amd64 \
+    CGO_ENABLED=0
+
+COPY go.mod go.sum ./
+RUN go mod download
+
+# source files
+COPY pkg ./pkg
```

Comment:
Normally copying files piecemeal like this rather than just the current directory would be suggested against.  In the context of a library, though, it makes sense here.